test/pkg/test/metadata.go

```diff
@@ -20,8 +20,12 @@ var (
 )
 
 func New[T client.Object](t T, mergeFrom ...T) T {
-	t.SetName(RandomName())
-	t.SetNamespace("default")
+	if t.GetName() == "" {
+		t.SetName(RandomName())
+	}
+	if t.GetNamespace() == "" {
+		t.SetNamespace("default")
```

Comment:
Just a note, we also need to test things in non-default namespace, e.g. 
* httproute 
* gateway